src/poprox_recommender/default.py

```diff
@@ -83,27 +83,22 @@ def personalized_pipeline(num_slots: int, algo_params: dict[str, Any] | None = N
         ranker = topk_ranker
 
     pipeline = RecommendationPipeline(name=diversify)
+
+    # Compute embeddings
     pipeline.add(article_embedder, inputs=["candidate"], output="candidate")
     pipeline.add(article_embedder, inputs=["clicked"], output="clicked")
     pipeline.add(user_embedder, inputs=["clicked", "profile"], output="profile")
+
+    # Score and rank articles with diversification/calibration reranking
     pipeline.add(article_scorer, inputs=["candidate", "profile"], output="candidate")
-    pipeline.add(topk_ranker, inputs=["candidate", "profile"], output="ranked")
     pipeline.add(ranker, inputs=["candidate", "profile"], output="reranked")
 
-    return pipeline
-
-
-def fallback_pipeline(num_slots: int) -> RecommendationPipeline:
-    """
-    Create the fallback (non-personalized) pipeline.
-
-    Args:
-        num_slots: The number of items to recommend.
-    """
-    topic_filter = TopicFilter()
-    sampler = UniformSampler(num_slots=num_slots)
+    # Output the plain descending-by-score ranking for comparison
+    pipeline.add(topk_ranker, inputs=["candidate", "profile"], output="ranked")
 
-    pipeline = RecommendationPipeline(name="random_topical")
+    # Fallback in case not enough articles came from the ranker
     pipeline.add(topic_filter, inputs=["candidate", "profile"], output="topical")
-    pipeline.add(sampler, inputs=["topical", "candidate"], output="ranked")
+    pipeline.add(sampler, inputs=["topical", "candidate"], output="sampled")
+    pipeline.add(fill, inputs=["reranked", "sampled"], output="recs")
```

Comment:
Once the pipeline itself supports fallbacks, we might consider whether we want to use `Fill` or a fallback here